### ISSUE TITLE
[bug]: python-mongo test mode flow breaking

### ISSUE THREAD

**Issue by nehagup** [keploy/keploy#709]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

During test mode for python application with mongo-flask as dependency, the test cases sometimes don't run, other times it runs and passes but overall report shows failed. It's intermittent when test cases run or not. 

### Steps to reproduce

1. Record test cases with this sample application https://github.com/keploy/samples-python/tree/main/flask-mongo 
2. Run test mode. See following error

```
    ▓▓▓▓██▓█▓▄
     ████████▓▒
          ▀▓▓███▄      ▄▄   ▄               ▌
         ▄▌▌▓▓████▄    ██ ▓█▀  ▄▌▀▄  ▓▓▌▄   ▓█  ▄▌▓▓▌▄ ▌▌   ▓
       ▓█████████▌▓▓   ██▓█▄  ▓█▄▓▓ ▐█▌  ██ ▓█  █▌  ██  █▌ █▓
      ▓▓▓▓▀▀▀▀▓▓▓▓▓▓▌  ██  █▓  ▓▌▄▄ ▐█▓▄▓█▀ █▓█ ▀█▄▄█▀   █▓█
       ▓▌                           ▐█▌                   █▌
        ▓

v0.1.0-dev

2023-08-17T14:05:33.501Z	INFO	cmd/test.go:61	🐰 Keploy:	{"keploy test and mock path": "/files/keploy", "keploy testReport path": "/files/keploy/testReports"}
2023-08-17T14:05:35.219Z	INFO	hooks/loader.go:629	🐰 Keploy:keploy initialized and probes added to the kernel.
2023-08-17T14:05:35.748Z	INFO	proxy/proxy.go:198	🐰 Keploy:Keploy has hijacked the DNS resolution mechanism, your application may misbehave in keploy test mode if you have provided wrong domain name in your application code.
2023-08-17T14:05:35.748Z	INFO	proxy/proxy.go:208	🐰 Keploy:Proxy started at port:16789
2023-08-17T14:05:35.748Z	INFO	proxy/proxy.go:392	🐰 Keploy:starting DNS server at addr:172.19.0.2:16789
[+] Running 2/0
 ✔ Container mongo      Created                                                                                                                                              0.0s
 ✔ Container flask-app  Created                                                                                                                                              0.0s
Attaching to flask-app, mongo
mongo      | {"t":{"$date":"2023-08-17T14:05:36.378+00:00"},"s":"I",  "c":"CONTROL",  "id":23285,   "ctx":"main","msg":"Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'"}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.378+00:00"},"s":"I",  "c":"NETWORK",  "id":4915701, "ctx":"main","msg":"Initialized wire specification","attr":{"spec":{"incomingExternalClient":{"minWireVersion":0,"maxWireVersion":17},"incomingInternalClient":{"minWireVersion":0,"maxWireVersion":17},"outgoing":{"minWireVersion":6,"maxWireVersion":17},"isInternalClient":true}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.380+00:00"},"s":"I",  "c":"NETWORK",  "id":4648601, "ctx":"main","msg":"Implicit TCP FastOpen unavailable. If TCP FastOpen is required, set tcpFastOpenServer, tcpFastOpenClient, and tcpFastOpenQueueSize."}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.381+00:00"},"s":"I",  "c":"REPL",     "id":5123008, "ctx":"main","msg":"Successfully registered PrimaryOnlyService","attr":{"service":"TenantMigrationDonorService","namespace":"config.tenantMigrationDonors"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.381+00:00"},"s":"I",  "c":"REPL",     "id":5123008, "ctx":"main","msg":"Successfully registered PrimaryOnlyService","attr":{"service":"TenantMigrationRecipientService","namespace":"config.tenantMigrationRecipients"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.381+00:00"},"s":"I",  "c":"REPL",     "id":5123008, "ctx":"main","msg":"Successfully registered PrimaryOnlyService","attr":{"service":"ShardSplitDonorService","namespace":"config.tenantSplitDonors"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.381+00:00"},"s":"I",  "c":"CONTROL",  "id":5945603, "ctx":"main","msg":"Multi threading initialized"}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.382+00:00"},"s":"I",  "c":"CONTROL",  "id":4615611, "ctx":"initandlisten","msg":"MongoDB starting","attr":{"pid":1,"port":27017,"dbPath":"/data/db","architecture":"64-bit","host":"5465712a39b4"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.382+00:00"},"s":"I",  "c":"CONTROL",  "id":23403,   "ctx":"initandlisten","msg":"Build Info","attr":{"buildInfo":{"version":"6.0.8","gitVersion":"3d84c0dd4e5d99be0d69003652313e7eaf4cdd74","openSSLVersion":"OpenSSL 3.0.2 15 Mar 2022","modules":[],"allocator":"tcmalloc","environment":{"distmod":"ubuntu2204","distarch":"aarch64","target_arch":"aarch64"}}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.382+00:00"},"s":"I",  "c":"CONTROL",  "id":51765,   "ctx":"initandlisten","msg":"Operating System","attr":{"os":{"name":"Ubuntu","version":"22.04"}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.382+00:00"},"s":"I",  "c":"CONTROL",  "id":21951,   "ctx":"initandlisten","msg":"Options set by command line","attr":{"options":{"net":{"bindIp":"*"}}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.383+00:00"},"s":"W",  "c":"STORAGE",  "id":22271,   "ctx":"initandlisten","msg":"Detected unclean shutdown - Lock file is not empty","attr":{"lockFile":"/data/db/mongod.lock"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.383+00:00"},"s":"I",  "c":"STORAGE",  "id":22270,   "ctx":"initandlisten","msg":"Storage engine to use detected by data files","attr":{"dbpath":"/data/db","storageEngine":"wiredTiger"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.383+00:00"},"s":"W",  "c":"STORAGE",  "id":22302,   "ctx":"initandlisten","msg":"Recovering data from the last clean checkpoint."}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.383+00:00"},"s":"I",  "c":"STORAGE",  "id":22297,   "ctx":"initandlisten","msg":"Using the XFS filesystem is strongly recommended with the WiredTiger storage engine. See http://dochub.mongodb.org/core/prodnotes-filesystem","tags":["startupWarnings"]}
mongo      | {"t":{"$date":"2023-08-17T14:05:36.383+00:00"},"s":"I",  "c":"STORAGE",  "id":22315,   "ctx":"initandlisten","msg":"Opening WiredTiger","attr":{"config":"create,cache_size=473M,session_max=33000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,remove=true,path=journal,compressor=snappy),builtin_extension_config=(zstd=(compression_level=6)),file_manager=(close_idle_time=600,close_scan_interval=10,close_handle_minimum=2000),statistics_log=(wait=0),json_output=(error,message),verbose=[recovery_progress:1,checkpoint_progress:1,compact_progress:1,backup:0,checkpoint:0,compact:0,evict:0,history_store:0,recovery:0,rts:0,salvage:0,tiered:0,timestamp:0,transaction:0,verify:0,log:0],"}}
2023-08-17T14:05:36.658Z	INFO	hooks/launch.go:127	🐰 Keploy:User application started successfully
2023-08-17T14:05:36.662Z	INFO	test/test.go:166	🐰 Keploy:	{"no of test cases": 3}
Answers[when resolution failed for query:1]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[after caching it]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[before appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[After appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 41508
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 A

;; ANSWER SECTION:
mongo.	3600	IN	A	172.19.0.2

🐰 Keploy:dns msg RCODE sending back:
0
flask-app  |  * Serving Flask app 'app' (lazy loading)
flask-app  |  * Environment: production
flask-app  |    WARNING: This is a development server. Do not use it in a production deployment.
flask-app  |    Use a production WSGI server instead.
Answers[when resolution failed for query:28]:
[]
Answers[after caching it]:
[]
Answers[before appending to msg]:
[]
Answers[After appending to msg]:
[]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 53285
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 AAAA

🐰 Keploy:dns msg RCODE sending back:
0
flask-app  |  * Debug mode: on
flask-app  | WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
flask-app  |  * Running on all addresses (0.0.0.0)
flask-app  |  * Running on http://127.0.0.1:6000
flask-app  |  * Running on http://172.19.0.4:6000
flask-app  | Press CTRL+C to quit
Answers[before appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[After appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 578
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 A

;; ANSWER SECTION:
mongo.	3600	IN	A	172.19.0.2

🐰 Keploy:dns msg RCODE sending back:
0
Answers[before appending to msg]:
[]
Answers[After appending to msg]:
[]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 53056
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 AAAA

🐰 Keploy:dns msg RCODE sending back:
0
flask-app  |  * Restarting with stat
Answers[before appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[After appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 34300
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 A

;; ANSWER SECTION:
mongo.	3600	IN	A	172.19.0.2

🐰 Keploy:dns msg RCODE sending back:
0
flask-app  |  * Debugger is active!
flask-app  |  * Debugger PIN: 547-785-906
Answers[before appending to msg]:
[]
Answers[After appending to msg]:
[]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 15330
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 AAAA

🐰 Keploy:dns msg RCODE sending back:
0
Answers[before appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[After appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 43583
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 A

;; ANSWER SECTION:
mongo.	3600	IN	A	172.19.0.2

🐰 Keploy:dns msg RCODE sending back:
0
Answers[before appending to msg]:
[]
Answers[After appending to msg]:
[]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 54590
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 AAAA

🐰 Keploy:dns msg RCODE sending back:
0
mongo      | {"t":{"$date":"2023-08-17T14:05:37.252+00:00"},"s":"I",  "c":"STORAGE",  "id":4795906, "ctx":"initandlisten","msg":"WiredTiger opened","attr":{"durationMillis":869}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.252+00:00"},"s":"I",  "c":"RECOVERY", "id":23987,   "ctx":"initandlisten","msg":"WiredTiger recoveryTimestamp","attr":{"recoveryTimestamp":{"$timestamp":{"t":0,"i":0}}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.256+00:00"},"s":"W",  "c":"CONTROL",  "id":22120,   "ctx":"initandlisten","msg":"Access control is not enabled for the database. Read and write access to data and configuration is unrestricted","tags":["startupWarnings"]}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.256+00:00"},"s":"W",  "c":"CONTROL",  "id":22178,   "ctx":"initandlisten","msg":"/sys/kernel/mm/transparent_hugepage/enabled is 'always'. We suggest setting it to 'never'","tags":["startupWarnings"]}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.256+00:00"},"s":"W",  "c":"CONTROL",  "id":5123300, "ctx":"initandlisten","msg":"vm.max_map_count is too low","attr":{"currentValue":65530,"recommendedMinimum":1677720,"maxConns":838860},"tags":["startupWarnings"]}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.258+00:00"},"s":"I",  "c":"NETWORK",  "id":4915702, "ctx":"initandlisten","msg":"Updated wire specification","attr":{"oldSpec":{"incomingExternalClient":{"minWireVersion":0,"maxWireVersion":17},"incomingInternalClient":{"minWireVersion":0,"maxWireVersion":17},"outgoing":{"minWireVersion":6,"maxWireVersion":17},"isInternalClient":true},"newSpec":{"incomingExternalClient":{"minWireVersion":0,"maxWireVersion":17},"incomingInternalClient":{"minWireVersion":17,"maxWireVersion":17},"outgoing":{"minWireVersion":17,"maxWireVersion":17},"isInternalClient":true}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.258+00:00"},"s":"I",  "c":"REPL",     "id":5853300, "ctx":"initandlisten","msg":"current featureCompatibilityVersion value","attr":{"featureCompatibilityVersion":"6.0","context":"startup"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.258+00:00"},"s":"I",  "c":"STORAGE",  "id":5071100, "ctx":"initandlisten","msg":"Clearing temp directory"}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.259+00:00"},"s":"I",  "c":"CONTROL",  "id":20536,   "ctx":"initandlisten","msg":"Flow Control is enabled on this deployment"}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.260+00:00"},"s":"I",  "c":"FTDC",     "id":20625,   "ctx":"initandlisten","msg":"Initializing full-time diagnostic data capture","attr":{"dataDirectory":"/data/db/diagnostic.data"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.263+00:00"},"s":"I",  "c":"REPL",     "id":6015317, "ctx":"initandlisten","msg":"Setting new configuration state","attr":{"newState":"ConfigReplicationDisabled","oldState":"ConfigPreStart"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.263+00:00"},"s":"I",  "c":"STORAGE",  "id":22262,   "ctx":"initandlisten","msg":"Timestamp monitor starting"}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.264+00:00"},"s":"I",  "c":"NETWORK",  "id":23015,   "ctx":"listener","msg":"Listening on","attr":{"address":"/tmp/mongodb-27017.sock"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.264+00:00"},"s":"I",  "c":"NETWORK",  "id":23015,   "ctx":"listener","msg":"Listening on","attr":{"address":"0.0.0.0"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:37.264+00:00"},"s":"I",  "c":"NETWORK",  "id":23016,   "ctx":"listener","msg":"Waiting for connections","attr":{"port":27017,"ssl":"off"}}
mongo      | {"t":{"$date":"2023-08-17T14:05:38.056+00:00"},"s":"I",  "c":"FTDC",     "id":20631,   "ctx":"ftdc","msg":"Unclean full-time diagnostic data capture shutdown detected, found interim file, some metrics may have been lost","attr":{"error":{"code":0,"codeName":"OK"}}}
2023-08-17T14:05:46.662Z	INFO	pkg/util.go:75	🐰 Keploy:making a http request	{"test case id": "test-1"}
Answers[before appending to msg]:
[]
Answers[After appending to msg]:
[]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 42097
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 AAAA

🐰 Keploy:dns msg RCODE sending back:
0
Answers[before appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
Answers[After appending to msg]:
[mongo.	3600	IN	A	172.19.0.2]
🐰 Keploy:dns msg sending back:
;; opcode: QUERY, status: NOERROR, id: 15995
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;mongo.	IN	 A

;; ANSWER SECTION:
mongo.	3600	IN	A	172.19.0.2

🐰 Keploy:dns msg RCODE sending back:
0
panic: runtime error: index out of range [1] with length 1

goroutine 52 [running]:
go.keploy.io/server/pkg/proxy/integrations/mongoparser.decodeOutgoingMongo(0x4001e672f8?, 0x9fb47f?, {0x4000142120, 0x118, 0x120}, {0xb53260?, 0x4001f0c020}, {0x0, 0x0}, 0x99a640?, ...)
	/app/pkg/proxy/integrations/mongoparser/mongoparser.go:244 +0x2ba0
go.keploy.io/server/pkg/proxy/integrations/mongoparser.ProcessOutgoingMongo(0x4000142120?, 0x9ec36b?, {0x4000142120, 0x118, 0x120}, {0xb53260, 0x4001f0c020}, {0x0, 0x0}, 0x4000a18195?, ...)
	/app/pkg/proxy/integrations/mongoparser/mongoparser.go:40 +0xe8
go.keploy.io/server/pkg/proxy.(*ProxySet).handleConnection(0x400265e1e0, {0xb532b8, 0x4001eb0008}, 0x0?)
	/app/pkg/proxy/proxy.go:699 +0x21e0
created by go.keploy.io/server/pkg/proxy.(*ProxySet).startProxy in goroutine 7
	/app/pkg/proxy/proxy.go:358 +0x73c
mongo      | {"t":{"$date":"2023-08-17T14:05:46.736+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.19.0.4:54114","uuid":"f33d6823-92cf-4651-8676-846681b011ef","connectionId":1,"connectionCount":1}}
mongo      | {"t":{"$date":"2023-08-17T14:05:46.737+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn1","msg":"client metadata","attr":{"remote":"172.19.0.4:54114","client":"conn1","doc":{"driver":{"name":"PyMongo","version":"4.4.1"},"os":{"type":"Linux","name":"Linux","architecture":"aarch64","version":"6.1.29-0-virt"},"platform":"CPython 3.9.17.final.0"}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:46.798+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.19.0.4:54120","uuid":"30c909aa-80fc-4df5-a0ab-a142e660bda5","connectionId":2,"connectionCount":2}}
mongo      | {"t":{"$date":"2023-08-17T14:05:46.799+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn2","msg":"client metadata","attr":{"remote":"172.19.0.4:54120","client":"conn2","doc":{"driver":{"name":"PyMongo","version":"4.4.1"},"os":{"type":"Linux","name":"Linux","architecture":"aarch64","version":"6.1.29-0-virt"},"platform":"CPython 3.9.17.final.0"}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:46.911+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.19.0.4:54122","uuid":"fd0f20f3-00e8-44a0-8b62-8cbfdc650b91","connectionId":3,"connectionCount":3}}
mongo      | {"t":{"$date":"2023-08-17T14:05:46.914+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn3","msg":"client metadata","attr":{"remote":"172.19.0.4:54122","client":"conn3","doc":{"driver":{"name":"PyMongo","version":"4.4.1"},"os":{"type":"Linux","name":"Linux","architecture":"aarch64","version":"6.1.29-0-virt"},"platform":"CPython 3.9.17.final.0"}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:47.237+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.19.0.4:54132","uuid":"715fdef0-3779-48df-9863-8190454aba88","connectionId":4,"connectionCount":4}}
mongo      | {"t":{"$date":"2023-08-17T14:05:47.237+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn4","msg":"client metadata","attr":{"remote":"172.19.0.4:54132","client":"conn4","doc":{"driver":{"name":"PyMongo","version":"4.4.1"},"os":{"type":"Linux","name":"Linux","architecture":"aarch64","version":"6.1.29-0-virt"},"platform":"CPython 3.9.17.final.0"}}}
mongo      | {"t":{"$date":"2023-08-17T14:05:47.240+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.19.0.4:54140","uuid":"b1e6f05e-a266-40df-8b22-05baf3458c0d","connectionId":5,"connectionCount":5}}
mongo      | {"t":{"$date":"2023-08-17T14:05:47.241+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn5","msg":"client metadata","attr":{"remote":"172.19.0.4:54140","client":"conn5","doc":{"driver":{"name":"PyMongo","version":"4.4.1"},"os":{"type":"Linux","name":"Linux","architecture":"aarch64","version":"6.1.29-0-virt"},"platform":"CPython 3.9.17.final.0"}}}
flask-app  | 172.19.0.2 - - [17/Aug/2023 14:05:47] "GET /students HTTP/1.1" 200 -
```

### Environment

Production

### Version

Local

### Repository

keploy

**Comment by yahayaohinoyi** [comment]
@nehagup can I assign to this?

**Comment by re-Tick** [comment]
Hi @nehagup, the issue was fixed in #820 for mocking the mongoDB responses. Logs for the testrun:
```bash
       ▓██▓▄
    ▓▓▓▓██▓█▓▄
     ████████▓▒
          ▀▓▓███▄      ▄▄   ▄               ▌
         ▄▌▌▓▓████▄    ██ ▓█▀  ▄▌▀▄  ▓▓▌▄   ▓█  ▄▌▓▓▌▄ ▌▌   ▓
       ▓█████████▌▓▓   ██▓█▄  ▓█▄▓▓ ▐█▌  ██ ▓█  █▌  ██  █▌ █▓
      ▓▓▓▓▀▀▀▀▓▓▓▓▓▓▌  ██  █▓  ▓▌▄▄ ▐█▓▄▓█▀ █▓█ ▀█▄▄█▀   █▓█
       ▓▌                           ▐█▌                   █▌
        ▓
  
v0.2.2-dev

🐰 Keploy: 2023-09-18T05:56:11Z         INFO    cmd/test.go:122         {"keploy test and mock path": "/files/keploy", "keploy testReport path": "/files/keploy/testReports"}
🐰 Keploy: 2023-09-18T05:56:14Z         INFO    hooks/loader.go:748     keploy initialized and probes added to the kernel.
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    proxy/proxy.go:389      Keploy has hijacked the DNS resolution mechanism, your application may misbehave in keploy test mode if you have provided wrong domain name in your application code.
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    proxy/proxy.go:450      Proxy started at port:16789
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    test/test.go:107        No testcases are recorded for the user application    {"for session": "test-set-0"}
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    proxy/proxy.go:640      starting DNS server at addr :16789
mockPath
/files/keploy/test-set-2
mocks
mockLen
10
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    hooks/launch.go:155     no error found while running user application container
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    hooks/launch.go:350     container & network found and processed successfully  {"time": 1695016575802946434}
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    hooks/launch.go:121     User application started successfully
🐰 Keploy: 2023-09-18T05:56:15Z         INFO    test/test.go:187                {"no of test cases": 5, "test-set": "test-set-2"}
 * Serving Flask app 'app' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:6000
 * Running on http://172.22.0.4:6000
Press CTRL+C to quit
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 143-800-723
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    pkg/util.go:65  making a http request   {"test case id": "test-1"}
172.22.0.3 - - [18/Sep/2023 05:56:25] "POST /students HTTP/1.1" 200 -
Testrun passed for testcase with id: "test-1"

--------------------------------------------------------------------

🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:232        result  {"testcase id": "test-1", "passed": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    pkg/util.go:65  making a http request   {"test case id": "test-2"}
Testrun passed for testcase with id: "test-2"

--------------------------------------------------------------------

🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:232        result  {"testcase id": "test-2", "passed": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    pkg/util.go:65  making a http request   {"test case id": "test-3"}
172.22.0.3 - - [18/Sep/2023 05:56:25] "GET /students HTTP/1.1" 200 -
172.22.0.3 - - [18/Sep/2023 05:56:25] "GET /students/12345 HTTP/1.1" 200 -
Testrun passed for testcase with id: "test-3"

--------------------------------------------------------------------

🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:232        result  {"testcase id": "test-3", "passed": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    pkg/util.go:65  making a http request   {"test case id": "test-4"}
Testrun passed for testcase with id: "test-4"

--------------------------------------------------------------------

🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:232        result  {"testcase id": "test-4", "passed": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    pkg/util.go:65  making a http request   {"test case id": "test-5"}
172.22.0.3 - - [18/Sep/2023 05:56:25] "PUT /students/12345 HTTP/1.1" 200 -
172.22.0.3 - - [18/Sep/2023 05:56:25] "DELETE /students/12345 HTTP/1.1" 200 -
Testrun passed for testcase with id: "test-5"

--------------------------------------------------------------------

🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:232        result  {"testcase id": "test-5", "passed": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:107        No testcases are recorded for the user application    {"for session": "test-set-1"}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    test/test.go:85 test run completed      {"passed overall": true}
🐰 Keploy: 2023-09-18T05:56:25Z         INFO    hooks/loader.go:365     Exiting keploy program gracefully.
🐰 Keploy: 2023-09-18T05:56:25Z         ERROR   hooks/ringBufReader.go:104      failed to receive signal from ringbuf socketDataEvent reader  {"error": "epoll wait: file already closed"}
🐰 Keploy: 2023-09-18T05:56:26Z         INFO    hooks/loader.go:414     eBPF resources released successfully...
🐰 Keploy: 2023-09-18T05:56:26Z         ERROR   proxy/proxy.go:596      failed to accept connection to the proxy      {"error": "accept tcp [::]:16789: use of closed network connection"}
🐰 Keploy: 2023-09-18T05:56:26Z         INFO    proxy/proxy.go:1079     Dns server stopped
🐰 Keploy: 2023-09-18T05:56:26Z         INFO    proxy/proxy.go:1081     proxy stopped...
```

**Comment by nehagup** [comment]
Thanks @re-Tick ! Closing this one :)
